pynsee/sirene/search_sirene.py

```diff
@@ -1,272 +1,291 @@
-        ):
-            patt = re.sub(r"\s+", "|", patt)
-        else:
-            # only single spaces are accepted in ranges
-            patt = re.sub(r"\s+", " ", patt)
-
-        list_patt = patt.split("|")
-
-        if upper_case:
-            list_var_patt_maj = [p.upper() for p in list_patt]
-            list_patt += list_var_patt_maj
-
-        if decode:
-            list_var_decode = [unidecode(p) for p in list_patt]
-            list_patt += list_var_decode
-
-        list_patt = list(set(list_patt))
-
-        list_var_patt = []
-        for ptt in list_patt:
-            if var in list_hist_variable:
-                ptt_h = "periode({}{}:{})".format(var, phntc_string, ptt)
-            else:
-                ptt_h = "{}{}:{}".format(var, phntc_string, ptt)
-
-            list_var_patt += [ptt_h]
-
-        list_var_pattern.append(list_var_patt)
-
-    if and_condition:
-        condition = " AND "
-    else:
-        condition = " OR "
-
-    permutation = list(itertools.product(*list_var_pattern))
-    permutation_and = [condition.join(list(x)) for x in permutation]
-    permutation_and_parenth = ["(" + x + ")" for x in permutation_and]
-    string_query = " OR ".join(permutation_and_parenth)
-
-    query = "?q=" + string_query
-    list_var_string = [str(b) for b in [kind, number]]
-    string = "".join(list_var_string)
-
-    data_final = _request_sirene(query=query, kind=kind, number=number)
-
-    df = _clean_data(
-        data_final.copy(),
-        kind=kind,
-        clean=False,
-        activity=activity,
-        legal=legal,
-        only_alive=alive,
-    )
-
-    if df is not None:
-        df = df.reset_index(drop=True)
-
-    _warning_search_sirene()
-
-    return df
+# -*- coding: utf-8 -*-
+# Copyright : INSEE, 2021
+
+from functools import lru_cache
+import itertools
+import re
+from unidecode import unidecode
+
+from pynsee.sirene._clean_data import _clean_data
+from pynsee.sirene._request_sirene import _request_sirene
+from pynsee.sirene.SireneDataFrame import SireneDataFrame
+from pynsee.utils.save_df import save_df
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@lru_cache(maxsize=None)
+def _warning_search_sirene():
+    logger.warning(
+        "This function may return personal data, please check and comply with "
+        "the legal framework relating to personal data protection !"
+    )
+
+
+@lru_cache(maxsize=None)
+def _warning_data_save():
+    logger.info(
+        "Locally saved data has been used\n"
+        "Set update=True to trigger an update"
+    )
+
+
+@save_df(day_lapse_max=30, obj=SireneDataFrame)
+def search_sirene(
+    variable,
+    pattern,
+    kind="siret",
+    phonetic_search=False,
+    and_condition=True,
+    upper_case=False,
+    decode=False,
+    number=1000,
+    activity=True,
+    legal=False,
+    closed=False,
+    update=False,
+    silent=False,
+):
+    """Get data about companies from criteria on variables

```

Comment:
@tgrandje this docstring seems to have been reformatted...

Reply:
Must be something I handled manually in the between, I'll reset that

Reply:
Yes, that's most probably an after effect of me removing the " #" trailing the code ; at that time flake8 was still blocking every commit because of the size of the docstrings...

Reply:
Done